### ISSUE TITLE
Code 1: The operation was cancelled.

### ISSUE THREAD

**Issue by draperunner** [googleapis/nodejs-redis#389]
I am trying to run my Node.js application with this client library, but it fails instantly with this error message:

```
{
  "code": 1,
  "details": "The operation was cancelled.",
  "metadata": {
    "internalRepr": {},
    "options": {}
  }
}
```

It's not very helpful. And is there a place I can find a reference of the error codes? I couldn't find anything in [the API reference](https://googleapis.dev/nodejs/redis/latest/index.html)

I have exported the GOOGLE_APPLICATION_CREDENTIALS env variable and double-checked that it's a valid service account with access to the Google Cloud Memorystore for Redis API.

I'm running the sample code below:
```
/**
 * List instances in a given location.
 * @param {string} project.
 * @param {string} location, e.g., us-east-1.
 */
async function main(projectId, location) {
    const { CloudRedisClient } = require('@google-cloud/redis')
    const client = new CloudRedisClient()
    const formattedParent = client.locationPath(projectId, location)
    const request = {
        parent: formattedParent,
    }
    try {
        const resp = (await client.listInstances(request))[0]
        console.info(resp)
    } catch (error) {
        console.error(error)
    }
}

main('<my-project-name>', '<my-procect-region>')
```

Do I need to set up a VPC connector if I'm using this SDK with a service account?

**Comment by bcoe** [comment]
@draperunner I wouldn't expect you need to connect through a VPC, are you running on your local network? and is there a proxy you connect through perhaps?

**Comment by draperunner** [comment]
Yes, I'm trying to run this on my computer on my local home network. I'm not using a proxy or anything like that. And my home office does not have any complex enterprise firewall setup either. A colleague suggested I would need to set up a proxy, but I hoped the service account credentials would be enough for local development.

After I found this: https://cloud.google.com/appengine/docs/standard/nodejs/using-memorystore I kinda gave up on this SDK and installed redis locally and the suggested `node_redis` SDK.

**Comment by crwilcox** [comment]
@draperunner I can't repro this with the code provied. I used my project and location `us-west2`. 

With no redis instances it returns an empty array.
With a single redis instance:
`Object {labels: Object, redisConfigs: Object, name: "projects/PROJECTNAME/locations/us-west2/instance…", displayName: "", locationId: "us-west2-b", …}`

**Comment by draperunner** [comment]
I'll see if I can debug some more tomorrow. I am using `us-central1` and Node.js v10.13.0.

**Comment by draperunner** [comment]
## Found it!

TL;DR: The error comes from an old version of the transitive dependency @grpc/grpc-js. `rm package-lock.json && npm install` fixed it...

I created a new directory and ran the sample code I posted in the original post (with project and region replaced of course). Only dependency: `@google-cloud/redis`. To my surprise it worked fine.

I then tried the exact same code, but now with the file lying in the real project. Now I got the same error as described above. Therefore the error had to be with some dependency. From the stack trace I could see that `@grpc/grpc-js` was reporting the error. The result of `npm list @grpc/grpc-js` in the faulty project was:

```
└─┬ @google-cloud/redis@1.6.0
  └─┬ google-gax@1.15.1
    └── @grpc/grpc-js@0.6.18
```

while in the clean project:

```
└─┬ @google-cloud/redis@1.6.0
  └─┬ google-gax@1.15.2
    └── @grpc/grpc-js@0.7.9
```

Using `npm list google-gax` I found that `@google-cloud/logging` was also depending on it:
```
├─┬ @google-cloud/logging-winston@3.0.6
│ └─┬ @google-cloud/logging@7.3.0
│   └── google-gax@1.15.1  deduped
└─┬ @google-cloud/redis@1.6.0
  └── google-gax@1.15.1
```

Deleting the `package-lock.json` and installing again with `npm install` updated the transitive dependencies and it works now. 🙄😄
